### DISCUSSION TITLE
Why it does not work?

### DISCUSSION THREAD

**Discussion by Serhioromano** [eclipse-langium/langium#792] [Q&A] [answered]
This works if I type `myname : INT;`

```
Declarations:
    variable_name=ID  ':' variable_type=VariableType ';'
;
VariableType returns string:
    'BOOL' | 'INT'
;
```

This does not

```
Declarations:
    variable_name=ID  ':' variable_type=VariableType ';'
;
VariableType returns string:
    'BOOL' | ANY_INT
;
terminal ANY_INT: 'U'? ('S'|'D'|'L')? 'INT';
```

It cannot resolve 

```
Expecting: one of these possible Token sequences:
  1. [BOOL:KW]
  2. [ANY_INT]
but found: 'INT' st(parsing-error)

**Comment by msujew** [comment, marked as answer]
Hey @Serhioromano,

do you already have a keyword or other terminal in your grammar which parses the `INT` string? Keywords are always preferred when lexing, so in case you encounter just `INT`, it will try to lex the `INT` keyword, not the `ANY_INT` terminal which lexes the same token.

**Reply by Serhioromano** [reply]
It was that simple! Thank you.
